README.md

```diff
@@ -237,8 +237,7 @@ You will find compiled binaries in `build-output/` respective to their source pa
 ```bash
 git submodule update --init
 export CMAKE_FLAGS=-DOPENSSL_ROOT_DIR=/usr/local/opt/libressl
-cmake -H. -Bbuild-output ${CMAKE_FLAGS}
-
+cmake -H. -Bbuild-output ${CMAKE_FLAGS?}
```

Comment:
This works as long as CMAKE_FLAGS is not set, not sure if it's going too far to check for `OPENSSL_ROOT_DIR` in the value.

I'm also a little confused by `cmake -H. -Bbuild-output ${CMAKE_FLAGS?}`. IIUC `CMAKE_FLAGS` is used whether a user uses `cmake -H. -Bbuild-output ${CMAKE_FLAGS?}` or `cmake -H. -Bbuild-output`. 
`cmake -H. -Bbuild-output ${CMAKE_FLAGS?}` is used to cache the value and could be used over a default value. 

Though not sure if there's a default for `OPENSSL_ROOT_DIR` to override in this case or maybe that's the point (just in case maybe)?

Reply:
So this is a bit of a hack, and it is hiding an ugly mess: we need to call cmake with `-DOPENSSL_ROOT_DIR=...` twice:

- Once to generate the Makefiles for `google-cloud-cpp/google/...` that is the call from the user.
- Next the CMake files call cmake again, to generate the Makefiles for `gRPC`.

Unfortunately the command-line flags provided by the user do not make it to that second call.  So we work around it by setting `CMAKE_FLAGS`.  Now that you asked me maybe a cleaner solution would be to set `OPENSSL_ROOT_DIR` in the environment.  That would have the desired effect in both calls to cmake, without the extra hacking.

Do you think we can merge this PR, which at least gives better error messages and clean this up later?

Reply:
Ah, okay, +1 for potential cleaner instruction of setting openssl_root_dir instead. It clear what's important in this case. Thanks for clearing that up. Approving this PR to get a better error message for now. 

Thanks @coryan!